src/pypeh/core/interfaces/outbound/dataops.py

```diff
@@ -130,17 +126,66 @@ def validate(
         return ret
 
 
-class DataImportInterface(OutDataOpsInterface, Generic[T_DataType]):
+class DateEnrichmentInterface(OutDataOpsInterface, Generic[T_DataType]):
+    @abstractmethod
+    def _enrich_data(self, data: dict[str, Sequence] | T_DataType, config: dict) -> T_DataType:
+        raise NotImplementedError
+
     @classmethod
     def get_default_adapter_class(cls):
         try:
-            adapter_module = importlib.import_module("pypeh.adapters.outbound.validation.pandera_adapter.dataops")
+            adapter_module = importlib.import_module("pypeh.adapters.outbound.enrichment.dataops")
             adapter_class = getattr(adapter_module, "DataFrameAdapter")
         except Exception as e:
-            logger.error("Exception encountered while attempting to import a Pandera-based DataFrameAdapter")
+            logger.error("Exception encountered while attempting to import enrichment DataFrameAdapter")
             raise e
         return adapter_class
 
+    @abstractmethod
+    def _join_data(self) -> T_DataType:
+        ...
+        # Add required variables to perform enrichment steps either from other observations or Mol, LOD, LOQ, SG-ref, etc...
+
+    def _build_calculation_plan_from_observable_properties(
```

Comment:
this is indeed a required step, although I am not sure what the calculation plan is. I am assuming this is for a single dataframe. So we'll have to structure this as `build_calculation_plan_for_observation`. See latest `load_tabular_data_collection` method in session.